src/parse.rs

```diff
@@ -0,0 +1,218 @@
+// SPDX-License-Identifier: CC0-1.0
+
+//! Hex encoding and decoding.
```

Comment:
No encoding, just decoding. Also I'd rename the module to `decode`. Feel free to leave to another PR.